scripts/test.sh

```diff
@@ -100,7 +100,7 @@ function run_unit_tests {
   local pkgs="${1:-./...}"
   shift 1
   # shellcheck disable=SC2086
-  GOLANG_TEST_SHORT=true go_test "${pkgs}" "parallel" : -short -timeout="${TIMEOUT:-3m}" "${COMMON_TEST_FLAGS[@]}" "${RUN_ARG[@]}" "$@"
+  GOLANG_TEST_SHORT=true go_test "${pkgs}" "parallel" : -p=2 -short -timeout="${TIMEOUT:-3m}" "${COMMON_TEST_FLAGS[@]}" "${RUN_ARG[@]}" "$@"
```

Comment:
Thanks for raising these ideas @Rajalakshmi-Girish.

I'm curious on the reasons behind setting parallel to `-p=2` across the board. From memory the default setting is `GOMAXPROCS` if not specified.  So are we saying that this default setting is causing problems and `-p=2` is somehow going to work better? If so I am keen to understand this more.

Reply:
@jmhbnz Can you please review now?